src/shred_collector/shred.zig

```diff
@@ -236,9 +292,34 @@ pub fn GenericShred(
             }
             return self.payload[SIGNATURE_LENGTH..end];
         }
+
+        fn verify(self: Self, signer: sig.core.Pubkey) bool {
+            const signed_data = self.signedData() catch return false;
+            const signature = layout.getSignature(self.payload) orelse return false;
+            return signature.verify(signer, &signed_data.data);
+        }
+
+        fn signedData(self: Self) !Hash {
+            return getMerkleRoot(self.payload, constants, self.common.shred_variant);
+        }
+
+        pub fn merkleRoot(self: Self) !Hash {
+            return getMerkleRoot(self.payload, constants, self.common.shred_variant);
+        }
```

Comment:
why do we have two methods that do the same thing?

Reply:
d9bd45a3b71e1178bb9ca9c8c395d22c8f17e38a